### ISSUE TITLE
Broken hyper/image links in external markdown files when base directory is different from html file

### ISSUE THREAD

**Issue by thomham** [hakimel/reveal.js#2734]
First of all - thank you for this great piece of software. 👍

## Problem

When using external markdown files with Reveal.js, the assumed base directory for each markdown file is the base directory of the HTML file.

When including several external markdown files from **different** locations, this is not correct. It leaves all **relative hyper links** and **relative image links** pointing to the wrong location.

The correct base directory for relative links in each markdown file should be the base directory of the HTML file concatenated with the relative path of the external markdown file.

Is there already a solution to this I do not see when having the markdown files in the same directory is not an option❓

*Your help is appreciated.* ❤️

## Example

Imagine the following file structure on the web server:

- `on-level-1.html`
- `on-level-1.md`
- `image/`
  - `robot-hand.jpg`
- `level-2/`
  - `on-level-2.md`

Content of `on-level-1.html`:

```html
...
<div class=“slides”>
    <section data-markdown="on-level-1.md"
                data-separator="^\s*\r?\n---\s*\r?\n"
                data-separator-vertical="^\s*\r?\n------\s*\r?\n"
                data-separator-notes="^\s*\r?\nNotes?:\s*\r?\n">
    </section>
    <section data-markdown="level-2/on-level-2.md"
                data-separator="^\s*\r?\n---\s*\r?\n"
                data-separator-vertical="^\s*\r?\n------\s*\r?\n"
                data-separator-notes="^\s*\r?\nNotes?:\s*\r?\n">
    </section>
</div>
...
```

Content of `on-level-1.md`:

```markdown
# On level 1
![robot hand](image/robot-hand.jpg)
```

Content of `on-level-2.md`:

```markdown
# On level 2
![robot hand](../image/robot-hand.jpg)
```

## Actual behavior

- Each markdown itself is correct and renders correct in e.g. VSCode or Github
- Reveal.js does not find `../image/robot-hand.jpg` in `on-level-2.md` as it assumes the same base directory as the `on-level-1.html` file.

## Expected behavior

Reveal.js should prefix each relative image link and each relative hyper link with the absolute path of the HTML file, here the absolute path of `on-level-1.html`.

**Comment by bhigy** [comment]
I agree it would be nice to have the option to use paths relative to the md file.
